src/_pytest/fixtures.py

```diff
@@ -1052,6 +1054,77 @@ def execute(self, request: SubRequest) -> _FixtureValue:
         result = hook.pytest_fixture_setup(fixturedef=self, request=request)
         return result
 
+    def _will_be_finalized_by_fixture(
+        self, fixturedef: Union["FixtureDef", PseudoFixtureDef]
```

Comment:
You would be correct. I'm updating it now. I was previously doing this because mypy complained, so I'm putting an explicit assert in so it knows not to expect a PseudoFixtureDef there